### DISCUSSION TITLE
Manual pre-release testing

### DISCUSSION THREAD

**Discussion by muety** [muety/wakapi#358] [General]
:wave: I'm looking for two or three people who are willing to test new pre-releases of Wakapi on their own, self-hosted instance.

**Background:** Wakapi features both unit- and API tests, but, admittedly, both are not super comprehensive. Some testing is still manual. Plus, over the last months and years of development, it has turned out that the biggest error sources in Wakapi are (a) the different databases supported and (b) database schema migrations. Often times, things work fine in my dev environment or on Wakapi.dev, but then fail at some of your self-hosted instances. These failures are often caused by using a different DBMS, operating system or CPU architecture ([...](https://github.com/muety/telepush/issues/40)), by upgrading from a different (older?) Wakapi version or simply by some rare edge case represented in your data. 

Before a new release that includes changes to the core logic, I test quite extensively on MySQL, using a dump from Wakapi.dev (only containing my own data). On Postgres and SQLite I only test very briefly.

It would be super helpful to me, if there was a small pool of Wakapi self-hosters (preferably with different databases), who - when asked to - do no more than **pull** the latest pre-release code, **compile** it, try to **run** it, briefly **click through** the dashboard and **report** back any errors that come up while doing so. Preferably, you'd backup your DB before, just in case :wink:.

If you like to participate, just drop a quick comment here and subscribe to the topic :point_down:.

**Comment by oasido** [comment]
Hey, I'm self-hosting a Wakapi instance only for myself, using the default database, I'd be more than happy to (try) to help.

**Comment by muety** [comment]
Awesome, thanks a lot in advance! :raised_hands: 

Anyone who's keen on testing right now, please do this:

1. Check out ff7d595a86fb324f3210e80a055b1ffca8b1bd7b
    ```bash
    git fetch
    git checkout ff7d595a86fb324f3210e80a055b1ffca8b1bd7b
    ```
1. Compile
    ```bash
    go build
    ```
1. Run the server
1. Bring up your IDE and open / close a couple of files in some project
1. Bring up the dashboard and browse through briefly
1. Report back: post a comment here, including:
    1. Status: 👌 or ❌ (for success or failure)
    3. Database: MySQL, SQLite, Postgres, ...
    4. OS: Win, Linux, Mac, ...
    5. Previous Wakapi version
    6. In case of ❌ failure: an error log or description of what went wrong

**Reply by YC** [reply]
i. Ok
ii. SQLite
iii. Linux
iv. 2.3.3 in Docker, db file pulled locally
